File: 30 Days of Code/Cpp/Day10_BinaryNumbers.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');
    int count, maxcount ;
    count = maxcount = 0;
    while(n){
        int rem = n%2;
        if(rem == 1){
            count++;
            maxcount = max(maxcount, count);
        }
        else {
            count = 0;
        }
        n = n/2;
    }
    
    
    cout << maxcount << endl;

    return 0;
}

File: 30 Days of Code/Java/Day10_BinaryNumbers.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Day10_BinaryNumbers{
    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
        int count , maxcount;
        count = maxcount =0;
        while(n!=0){
            int rem = n%2;
            if (rem == 1){
                count++;
                maxcount = Math.max(maxcount, count);
            }else 
            count = 0;
            n = n/2;
        }
        System.out.println(maxcount);
        scanner.close();
    }
}
